web-client/src/lib.rs

```diff
@@ -174,11 +177,36 @@ impl WebClient {
 
         loop {
             match network_events.next().await {
-                Some(Ok(NetworkEvent::PeerJoined(peer_id))) => {
-                    peer_listener("joined", peer_id.to_string(), network.peer_count());
+                Some(Ok(NetworkEvent::PeerJoined(peer_id, peer_info))) => {
+                    let node_type = if peer_info
+                        .get_services()
+                        .contains(Services::provided(NodeType::History))
+                    {
+                        "History"
+                    } else if peer_info
+                        .get_services()
+                        .contains(Services::provided(NodeType::Full))
+                    {
+                        "Full"
+                    } else {
+                        "Light"
+                    };
+                    peer_listener(
+                        "joined",
+                        peer_id.to_string(),
+                        peer_info.get_address().to_string(),
+                        node_type,
+                        network.peer_count(),
+                    );
                 }
                 Some(Ok(NetworkEvent::PeerLeft(peer_id))) => {
-                    peer_listener("left", peer_id.to_string(), network.peer_count());
+                    peer_listener(
+                        "left",
+                        peer_id.to_string(),
+                        String::from(""),
+                        "",
+                        network.peer_count(),
+                    );
```

Comment:
@sisou Is there a better way to do this? Now the _join_ event carries more information than the _left_ event. I thought on separating the listener for _join_ and _left_ events but not sure if that's what we want.

Reply:
After thinking about it, my suggestion would be to not pass the three peer info bites individually, but to pass a `PeerInfo` object (either as JSON, or as a bridged struct like in my other branch). Then everything but the peer ID can be an optional field.